feat: add filename sorting

Computerdores opened the pull request:
### Summary
Aims to add sorting by filename in order to close #822 and address the roadmap item.

### Tasks Completed

-   Platforms Tested:
    -   [ ] Windows x86
    -   [ ] Windows ARM
    -   [ ] macOS x86
    -   [ ] macOS ARM
    -   [ ] Linux x86
    -   [ ] Linux ARM
-   Tested For:
    -   [ ] Basic functionality
    -   [ ] PyInstaller executable

Computerdores commented:
this is now ready for review

CyanVoxel commented:
I apologize for the inconvenience, #844 has now been pulled and a rebase targeting `main` will be required here now.
Thank you for your work on this so far, and sorry again for the disruption!

Computerdores commented:
rebase is done. I have also fixed the search test by updating the search library to the latest db version. Further I have added a db migration test for the new migration

CyanVoxel reviewed (approved):
Everything looks good here, the only rough spot I've run into is that it can take some time for the new filename column to populate when opening a library (45 seconds for 200k files on an SSD for me).

It may be a good idea (in a future PR if antyhing) to reformat the SQL to SQL migration logic in a way similar to how the JSON to SQL logic checks in with the frontend first before migrating, just to allow for something like an indefinite progress bar popup notifying the user that things are still happening. Though this process also took place during the splash screen for me since I used the launch flags, so maybe there's even more than needs to be considered first.

In any case, thank you so much for your work on adding this! And thank you for updating the test files as well!
